covid19dp_submission/download_analyses.py

```diff
@@ -127,6 +140,34 @@ def download_files(analyses_array, download_target_dir, processed_analyses_file)
                     os.remove(download_file_path)
 
 
+@retry(exceptions=(UnfinishedBatchError,), logger=logger, tries=4, delay=10, backoff=1.2, jitter=(1, 3))
+def download_files_via_aspera(analyses_array, download_target_dir, processed_analyses_file, ascp, aspera_id_dsa,
+                              batch_size=100):
+    logger.info(f"total number of files to download: {len(analyses_array)}")
+    remaining_analyses_array = []
+    with open(processed_analyses_file, 'a') as open_file:
+        for analysis_batch in chunked(analyses_array, batch_size):
+            download_urls = [
+                f"era-fasp@{analysis['submitted_aspera']}" for analysis in analysis_batch if analysis
+            ]
+            command = f'{ascp} -i {aspera_id_dsa} -QT -l 300m -P 33001 {" ".join(download_urls)} {download_target_dir}'
+            run_command_with_output(f"Download batch of covid19 data", command)
```

Comment:
I think you're onto something. I implemented the retry mechanisms because I was not getting all the files at the end but it never actually failed (probably due to the bug I fixed [here](https://github.com/EBIvariation/covid19dp-submission/pull/12/commits/ac0cc47e9fe0be2e20e01cfa480ab32421484aaf)). I just checked and `ascp` does return non 0 exit status if any file is missing.
I now think that the retry mechanism is not going to be useful at all.

Reply:
Just as I wrote this comment I check an actual execution and saw evidence of aspera successfully finishing with only 99/100 files (see `cat ingest_covid19dp.log  | grep -B3 Failed` in the cluster). There is something I'm probably missing but the retry mechanism would kick in.